### PR TITLE
[hackathon] Data Pipeline to read Criteo TSV and Parquet files

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #183

This PR is a demo and will NOT be merged.

### CODE REVIEW

File: examples/hackathon/critero_tsv.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+map_dp = file_dp.map(get_name)
+csv_parser_dp = map_dp.parse_csv(delimiter="\t", skip_lines=1)
+columns_names = list(map_dp.parse_csv(delimiter="\t"))[0][1:]
+processed_dp = csv_parser_dp.map(row_pre_process)
+header_dp = processed_dp.header(1000)  # limit to 1000 rows to save time
+# TODO: The follow operation is very slow due to torcharrow.DataFrame
+#       It gives the following warning:
+#       "UserWarning: append for type NumericalColumnCpu is suported only with prototype implementation,
+#        which may result in degenerated performance"
+dfs = header_dp.dataframe(dataframe_size=1000, dtype=DTYPE)
```
Comment (NivekT) [inline examples/hackathon/critero_tsv.py:50]: @VitalyFedyunin This is the one you should use

File: examples/hackathon/criteo_parquet.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+DTYPE = dt.Struct(
+    [dt.Field("label", dt.int64)]
+    + [dt.Field(int_name, dt.Float64(nullable=True)) for int_name in DEFAULT_INT_NAMES]
+    + [dt.Field(cat_name, dt.Float64(nullable=True)) for cat_name in DEFAULT_CAT_NAMES]
+)
+
+source_dp = FileLister(".", masks="*.parquet")
+# TODO: There is current a bug in TorchArrow/Velox related to conversion of arrow arrys to DataFrame that causes
```
Comment (NivekT) [inline examples/hackathon/criteo_parquet.py:17]: Waiting on fix from Velox/TorchArrow - https://github.com/facebookresearch/torcharrow/issues/146

File: examples/hackathon/critero_tsv.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+)
+
+source_dp = FileLister(".", masks="day_11_first_3k_rows.tsv")
+file_dp = FileOpener(source_dp, mode="b")
+map_dp = file_dp.map(get_name)
+csv_parser_dp = map_dp.parse_csv(delimiter="\t", skip_lines=1)
+columns_names = list(map_dp.parse_csv(delimiter="\t"))[0][1:]
+processed_dp = csv_parser_dp.map(row_pre_process)
+header_dp = processed_dp.header(1000)  # limit to 1000 rows to save time
+# TODO: The follow operation is very slow due to torcharrow.DataFrame
```
Comment (NivekT) [inline examples/hackathon/critero_tsv.py:46]: The `.dataframe` operation is slow as the warning indicates.

cc: @wenleix

Comment (wenleix) [reply examples/hackathon/critero_tsv.py:46]: Is the input list of tuple -- should be fixed via https://github.com/facebookresearch/torcharrow/pull/159

Comment (NivekT) [reply examples/hackathon/critero_tsv.py:46]: Yes, the input is a list of tuple.
```python
print(list(header_dp)[0])  # This is a tuple (0.0, 118.0, ...)
```
Then in `.dataframe`, `batch` and `map` are called:
```python
batch_dp = source_dp.batch(dataframe_size)
# I just confirmed that the line of code below is the one that is very slow
df_dp = batch_dp.map(partial(torcharrow.DataFrame, dtype=dtype, columns=columns, device=device))  
return df_dp
```

Comment (wenleix) [reply examples/hackathon/critero_tsv.py:46]: @NivekT Can you try TorchArrow with https://github.com/facebookresearch/torcharrow/pull/159 ? (BTW did you build TorchArrow locally ? :) )

Comment (NivekT) [reply examples/hackathon/critero_tsv.py]: I just updated TorchArrow and rebuilt. I can confirm that it is 100x faster. Thanks!

**Comment by VitalyFedyunin** [comment]
Somehow `ghstack` overwrote your PR. Sorry.

**Comment by NivekT** [comment]
> Somehow `ghstack` overwrote your PR. Sorry.

I think it is possible to revert if needed. 

https://github.com/pytorch/data/pull/183/commits

**Comment by NivekT** [comment]
Closing this PR since the changes have been separately landed

**Pull request closed without merging**